File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

class Main{
    static int matrix[][];
    static int v, e;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        v = Integer.parseInt(br.readLine());
        e = Integer.parseInt(br.readLine());
        matrix=new int[v+1][v+1];
        for (int i = 0; i < e; i++) {
            String s = br.readLine();
            StringTokenizer st=new StringTokenizer(s);
            int a= Integer.parseInt(st.nextToken());
            int b= Integer.parseInt(st.nextToken());
            matrix[a][b]=1;
            matrix[b][a]=1;
        }
        int result=bfs(1);
        System.out.println(result);
    }

    public static int bfs(int startx) {
        int cnt=0;
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(startx);
        int visited[]=new int[v+1];
        visited[startx]=1;
        while (!queue.isEmpty()){
            int now=queue.poll();
            for (int i = 1; i <=v ; i++) {
                if(matrix[now][i]==1&&visited[i]!=1){
                    queue.offer(i);
                    visited[i]=1;
                    cnt++;
                }
            }
        }
        return cnt;
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.cs
using System;
using System.Collections.Generic;

class Program
{
    private static int v, e;
    private static List<int>[] graph;

    public static void Main(string[] args)
    {
        v = int.Parse(Console.ReadLine());
        e = int.Parse(Console.ReadLine());

        graph = new List<int>[v + 1];

        for (int i = 0; i <= v; i++)
        {
            graph[i] = new List<int>();
        }

        for (int i = 0; i < e; i++)
        {
            var strings = Console.ReadLine().Split(" ");
            int a = int.Parse(strings[0]);
            int b = int.Parse(strings[1]);

            graph[a].Add(b);
            graph[b].Add(a);
        }
        
        Console.Write(BFS(1) - 1);
    }

    static int BFS(int start)
    {
        Queue<int> queue = new Queue<int>();
        queue.Enqueue(start);
        bool[] visited = new bool[v + 1];

        visited[start] = true;
        int cnt = 0;
        while (queue.Count > 0)
        {
            var now = queue.Dequeue();
            cnt++;

            for (int i = 0; i < graph[now].Count; i++)
            {
                var next = graph[now][i];
                if (visited[next]) continue;

                queue.Enqueue(next);
                visited[next] = true;
            }
        }

        return cnt;
    }
}